src/main/scala/junctions/asyncqueue.scala

```diff
@@ -2,22 +2,28 @@
 
 package junctions
 import Chisel._
+import uncore.util.{AsyncResetRegVec, AsyncResetReg}
 
 object GrayCounter {
   def apply(bits: Int, increment: Bool = Bool(true)): UInt = {
-    val binary = RegInit(UInt(0, width = bits))
-    val incremented = binary + increment.asUInt()
-    binary := incremented
+    val incremented = Wire(UInt(width=bits))
+    val binary = AsyncResetReg(incremented, 0, Bool(true))
+    incremented := binary + increment.asUInt()
     incremented ^ (incremented >> UInt(1))
   }
 }
 
 object AsyncGrayCounter {
   def apply(in: UInt, sync: Int): UInt = {
-    val syncv = RegInit(Vec.fill(sync){UInt(0, width = in.getWidth)})
-    syncv.last := in
-    (syncv.init zip syncv.tail).foreach { case (sink, source) => sink := source }
-    syncv(0)
+    val syncv = List.fill(sync)(Module (new AsyncResetRegVec(w = in.getWidth, 0)))
+    syncv.last.io.d := in
+    syncv.last.io.en := Bool(true)
+      (syncv.init zip syncv.tail).foreach { case (sink, source) => {
+        sink.io.d := source.io.q
+        sink.io.en := Bool(true)
+      }
+      }
+    syncv(0).io.d
```

Comment:
Yeah. I tried to think of some Scala fanciness, then decided sometimes you should just write what you mean.